src/backend/chat/custom/langchain.py

```diff
@@ -25,43 +39,27 @@ def __init__(self) -> None:
         # Prompt template
         prompt = ChatPromptTemplate.from_template("{input}")
 
-        python_tool = AVAILABLE_TOOLS[ToolName.Python_Interpreter].implementation()
-        tavily_search_tool = AVAILABLE_TOOLS[
-            ToolName.Tavily_Internet_Search
-        ].implementation()
+        tools = []
+        for req_tool in chat_request.tools:
+            tool = AVAILABLE_TOOLS.get(req_tool.name)
+            if tool:
+                tools.append(tool.implementation().to_langchain_tool())
+            else:
+                raise ValueError(f"Tool {req_tool.name} not found")
+
         # Create the ReAct agent
```

Comment:
```suggestion
        # Create the React agent
```

Reply:
It's actually ReAct haha

Reply:
https://python.langchain.com/docs/modules/agents/agent_types/react/